docs/guide/testing.md

```diff
@@ -162,7 +162,150 @@ You can generally solve this by calling [`pause()`][textual.pilot.Pilot.pause] w
+If there's a mismatch, the snapshot testing framework flags it for review.
+
+This offers a simple, automated way of checking our application displays like we expect.
+
+### pytest-textual-snapshot
+
+You can use [`pytest-textual-snapshot`](https://github.com/Textualize/pytest-textual-snapshot) to snapshot test your Textual app.
+This is a plugin for pytest which adds support for snapshot testing Textual apps, and it's maintained by the developers of Textual.
+
+A test using this package saves a snapshot (in this case, an SVG screenshot) of a running Textual app to disk.
+The next time the test runs, it takes another snapshot and compares it to the previously saved one.
+If the snapshots differ, the test fails, and you can view side-by-side diff showing the visual change.
+
+#### Installation
+
+You can install `pytest-textual-snapshot` using your favorite package manager (`pip`, `poetry`, etc.).
+
+```
+pip install pytest-textual-snapshot
+```
+
+#### Creating a snapshot test
+
+With the package installed, you now have access to the `snap_compare` pytest fixture.
+
+Let's look at an example of how we'd create a snapshot test for the [calculator app](https://github.com/Textualize/textual/blob/main/examples/calculator.py) below.
+
+```{.textual path="examples/calculator.py" columns=100 lines=41 press="3,.,1,4,5,9,2,wait:400"}
+```
+
+First, we need to create a new test and specify the path to the Python file containing the app.
+This path should be relative to the location of the test.
+
+```python
+def test_calculator(snap_compare):
+    assert snap_compare("path/to/calculator.py")
+```
+
+Let's run the test as normal using `pytest`.
+
+```
+pytest
+```
+
+When this test runs for the first time, an SVG screenshot of the calculator app is generated, and the test will fail.
+Snapshot tests always fail on the first run, since there's no previous version to compare the snapshot to.
+
+![snapshot_report_console_output.png](../images/testing/snapshot_report_console_output.png)
+
+If you open the snapshot report in your browser, you'll see something like this:
+
+![snapshot_report_example.png](../images/testing/snapshot_report_example.png)
+
+!!! tip
+
+    You can usually open the link directly from the terminal, but some terminal emulators may
+    require you to hold ++ctrl++ or ++command++ while clicking for links to work.
+
+The report explains that there's "No history for this test".
+It's our job to validate that the initial snapshot looks correct before proceeding.
+Our calculator is rendering as we expect, so we'll save this snapshot:
+
+```
+pytest --snapshot-update
+```
+
+!!! warning
+
+    Only ever run pytest with `--snapshot-update` if you're happy with how the output looks
+    on the left hand side of the snapshot report. When using `--snapshot-update`, you're saying "I'm happy with this
+    screenshot, and it will now represent the ground truth which all future runs of this test should be compared against".
```

Comment:
I think this is a key warning. Also, it may be worth putting extra emphasis on the fact that this will happen for _all_ snapshot tests that are already in place, so if any of those was wrong, it'll be updated in the wrong way.

Reply:
Good point - I've made that clearer